client/finality-grandpa/src/environment.rs

```diff
@@ -377,17 +377,17 @@ impl<Block: BlockT> SharedVoterSetState<Block> {
 }
 
 /// The environment we run GRANDPA in.
-pub(crate) struct Environment<B, E, Block: BlockT, N: NetworkT<Block>, RA, SC, VR> {
-	pub(crate) client: Arc<Client<B, E, Block, RA>>,
-	pub(crate) select_chain: SC,
-	pub(crate) voters: Arc<VoterSet<AuthorityId>>,
-	pub(crate) config: Config,
-	pub(crate) authority_set: SharedAuthoritySet<Block::Hash, NumberFor<Block>>,
-	pub(crate) consensus_changes: SharedConsensusChanges<Block::Hash, NumberFor<Block>>,
-	pub(crate) network: crate::communication::NetworkBridge<Block, N>,
-	pub(crate) set_id: SetId,
-	pub(crate) voter_set_state: SharedVoterSetState<Block>,
-	pub(crate) voting_rule: VR,
+pub struct Environment<B, E, Block: BlockT, N: NetworkT<Block>, RA, SC, VR> {
+	pub client: Arc<Client<B, E, Block, RA>>,
```

Comment:
Do we need to make the fields public? We only care about exporting the `Environment` type. If the fields are not public I assume that a lot less types would have to be exported.

Reply:
Yeah I have some hope to be able to roll back some of these `pub` changes once it's all up and working